ethereum_history_api/circuits/lib/src/verifiers/tx_helpers/idx.nr

```diff
@@ -9,7 +10,8 @@ global MAX_TX_IDX_BYTES_LEN = 2;
 pub fn assert_tx_idx_equals(key_as_rlp: [u8; KEY_NIBBLE_LEN], tx_idx: Field) {
     let (key_right_padded, shifted_by) = left_to_right_padding(key_as_rlp);
     let key_bytes: [u8; KEY_NIBBLE_LEN / 2] = nibbles_to_bytes(key_right_padded);
-    let (rlp_offset, rlp_len) = decode_string(key_bytes);
+    let key_bytes_fragment = Fragment::new_focused(key_bytes);
+    let (rlp_offset, rlp_len) = decode_string(key_bytes_fragment);
```

Comment:
Why do we create key_bytes_fragment instead of doing conversion inline here?

Reply:
Because we later use it when calling `sub_array_equals_up_to_length`